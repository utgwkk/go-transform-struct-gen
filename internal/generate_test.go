package internal

import (
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestGenerate(t *testing.T) {
	t.Parallel()
	testcases := []struct {
		name      string
		dstStruct string
		srcStruct string
		opts      *GenerateOption
	}{
		{
			name:      "without createdAt, updatedAt (function)",
			dstStruct: "*github.com/utgwkk/go-transform-struct-gen/internal/fixtures/foo.FooModel",
			srcStruct: "*github.com/utgwkk/go-transform-struct-gen/internal/fixtures/bar.BarModel",
			opts: &GenerateOption{
				TransformerType: TransformerTypeFunction,
				TramsformerName: "NewFooFromBar",
				DestinationPath: "./fixtures/bar/to_foo.go",
			},
		},
		{
			name:      "without createdAt, updatedAt (method)",
			dstStruct: "*github.com/utgwkk/go-transform-struct-gen/internal/fixtures/foo.FooModel",
			srcStruct: "*github.com/utgwkk/go-transform-struct-gen/internal/fixtures/bar.BarModel",
			opts: &GenerateOption{
				TransformerType: TransformerTypeMethod,
				TramsformerName: "ToFoo",
				DestinationPath: "./fixtures/bar/to_foo.go",
			},
		},
		{
			name:      "without createdAt, updatedAt (method, no ref)",
			dstStruct: "github.com/utgwkk/go-transform-struct-gen/internal/fixtures/foo.FooModel",
			srcStruct: "*github.com/utgwkk/go-transform-struct-gen/internal/fixtures/bar.BarModel",
			opts: &GenerateOption{
				TransformerType: TransformerTypeMethod,
				TramsformerName: "ToFoo",
				DestinationPath: "./fixtures/bar/to_foo.go",
			},
		},
	}
	for _, tc := range testcases {
		tc := tc
		t.Run(tc.name, func(t *testing.T) {
			t.Parallel()
			dst, err := ResolveStruct(tc.dstStruct)
			require.NoError(t, err)
			src, err := ResolveStruct(tc.srcStruct)
			require.NoError(t, err)

			got, err := Generate(dst, src, tc.opts)
			require.NoError(t, err)

			snaps.MatchSnapshot(t, got)
		})
	}
}
