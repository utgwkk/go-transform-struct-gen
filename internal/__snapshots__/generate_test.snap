
[TestGenerate/without_createdAt,_updatedAt_(function) - 1]
package bar

import "github.com/utgwkk/go-transform-struct-gen/internal/fixtures/foo"

func NewFooFromBar(src *BarModel) *foo.FooModel {
    return &foo.FooModel{
        Age:  src.Age,
        Id:   src.Id,
        Name: src.Name,
    }
}

---

[TestGenerate/without_createdAt,_updatedAt_(method) - 1]
package bar

import "github.com/utgwkk/go-transform-struct-gen/internal/fixtures/foo"

func (src *BarModel) ToFoo() *foo.FooModel {
    return &foo.FooModel{
        Age:  src.Age,
        Id:   src.Id,
        Name: src.Name,
    }
}

---

[TestGenerate/without_createdAt,_updatedAt_(method,_no_ref) - 1]
package bar

import "github.com/utgwkk/go-transform-struct-gen/internal/fixtures/foo"

func (src *BarModel) ToFoo() foo.FooModel {
    return foo.FooModel{
        Age:  src.Age,
        Id:   src.Id,
        Name: src.Name,
    }
}

---
